### PR TITLE
dbt docs

### PR DESCRIPTION
**Pull request by gingerwizard**
The plugin also supports snapshots but i've removed for now as there is an issue I need chase up and resolve.

### CODE REVIEW

File: docs/en/integrations/dbt/dbt-intro.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+sidebar_label: Introduction 
+sidebar_position: 1
+description: Users can transform and model their data in ClickHouse using dbts
+---
+
+# ClickHouse and dbt
+
+dbt (data build tool) enables analytics engineers to transform data in their warehouses by simply writing select statements. dbt handles materializing these select statements into objects in the database in the form of tables and views - performing the T of ELT. Users can create a model defined by a SELECT statement.
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-intro.md:9]: ELT is a newish acronym in the industry. We can detail Extract Load and Transform and maybe link a wikipedia page or something

File: docs/en/integrations/dbt/dbt-setup.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Install the dbt ClickHouse plugin:
+
+```bash
+pip install dbt-clickhouse
+```
+
+# Prepare ClickHouse
+
+dbt excels when modeling highly relational data. For the purposes of example, we provide a small IMDB dataset with the following relational schema. This dataset originates from the[ relational dataset repository](https://relational.fit.cvut.cz/dataset/IMDb). This is trivial relative to common schemas used with dbt but represents a manageable sample:
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-setup.md:31]: The dataset providers require to cite this paper when using their data:
https://arxiv.org/abs/1511.03086
see: [How to cite](https://relational.fit.cvut.cz/about)

File: docs/en/integrations/dbt/dbt-setup.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    role       String
+) ENGINE = MergeTree ORDER BY (actor_id, movie_id);
+```
+
+Note the column created_at for the table roles, which defaults to a value of now(). We use this later to identify incremental updates to our models - see [Incremental Models](./dbt-incremental-model).
+
+We use the s3 function to read the source data from public endpoints to insert data into these. Use the following to populate the tables:
+
+```sql
+INSERT INTO imdb.actors
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-setup.md:93]: The S3 repositories are not publicly available
```
Received exception from server (version 22.4.1):
Code: 499. DB::Exception: Received from localhost:9000. DB::Exception: Access Denied: While executing TabSeparatedRowInputFormat: While executing S3. (S3_ERROR)
```

File: docs/en/integrations/dbt/dbt-setup.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
++------+------------+----------+------------------+-------------+--------------+-------------------+
+|45332 |Mel Blanc   |832       |6.175853582979779 |18           |84            |2022-04-26 14:01:45|
+|621468|Bess Flowers|659       |5.57727638854796  |19           |293           |2022-04-26 14:01:46|
+|372839|Lee Phelps  |527       |5.032976449684617 |18           |261           |2022-04-26 14:01:46|
+|283127|Tom London  |525       |2.8721716524875673|17           |203           |2022-04-26 14:01:46|
+|356804|Bud Osborne |515       |2.0389507108727773|15           |149           |2022-04-26 14:01:46|
++------+------------+----------+------------------+-------------+--------------+-------------------+
+```
+
+In the later guides, we will convert this query into a model- materializing it in ClickHouse as a dbt view and table.
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-setup.md:165]: Missing space next to: model-

File: docs/en/integrations/dbt/dbt-connecting.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    https://docs.getdbt.com/docs/configure-your-profile
+    ```
+
+2. `cd` into your project folder.
+
+    ```bash
+    cd imdb
+    ```
+
+3. At this point, you will need the text editor of your choice. In the examples below, we use the popular VSCode. Opening the IMDB directory, you should see a collection of yml and sql files.
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-connecting.md:37]: For some reason, the screenshot images in this section are blurred and low quality. It's better to replace them with code blocks. For the folder structure, you can use the output of a `tree` command for eg.

File: docs/en/integrations/dbt/dbt-incremental-model.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    |372839|Lee Phelps  |527       |5.032976449684617 |18    |261      |2022-04-26 15:26:56|
+    |283127|Tom London  |525       |2.8721716524875673|17    |203      |2022-04-26 15:26:56|
+    |356804|Bud Osborne |515       |2.0389507108727773|15    |149      |2022-04-26 15:26:56|
+    +------+------------+----------+------------------+------+---------+-------------------+
+    ```
+
+3. We will now add data to our model to illustrate an incremental update. Add our actor  "Clicky McClickHouse" to the `actors` table.
+
+    ```sql
+        INSERT INTO imdb.actors VALUES (845466, 'Clicky', 'McClickHouse', 'M');
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-incremental-model.md:63]: I'm a huge fan of Clicky McClickHouse's movies. Glad to finally see some recognition for his work!

Seriously, we need to reuse this name everywhere in the docs :) cc @tom-clickhouse

File: docs/en/integrations/dbt/dbt-limitations.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+description: Limitations of the dbt ClickHouse plugin
+---
+
+# Limitations
+
+The current ClickHouse plugin for dbt has several limitations users should be aware of:
+
+
+
+1. As noted in [Creating an Incremental Materialization](./dbt-incremental-model#internals), incremental changes are currently loaded into an in-memory table. ClickHouse does not provide distribution for this table, so changesets must not be larger than the memory of the orchestrator node receiving the query. Users should either avoid using the plugin on datasets where changes/additions are very high or schedule this operation to run at appropriate intervals to ensure all changes can be captured within memory limits. As noted[ here](https://clickhouse.com/docs/en/engines/table-engines/special/memory), we recommend that in-memory table engines should not exceed 100,000,000 rows. Use this as an upper bound and schedule dbt run’s accordingly, i.e., ensure that the execution interval is frequent enough to not exceed this limit.
```
Comment (Ryado) [inline docs/en/integrations/dbt/dbt-limitations.md:13]: for readability, can you replace:
should not exceed 100,000,000 rows.
by
should not exceed 100 Million rows.

**Review by Ryado — COMMENTED** [review]
Great docs! added some comments.

Out of scope but maybe nice to add some words about "data as code" and the ability of versioning the dbt project (which is one of the main advantages of this approach IMO)

File: docs/en/integrations/dbt/dbt-setup.md (hunk 3 above)
Comment (gingerwizard) [reply docs/en/integrations/dbt/dbt-setup.md:93]: resolved

File: docs/en/integrations/dbt/dbt-incremental-model.md (hunk 6 above)
Comment (gingerwizard) [reply docs/en/integrations/dbt/dbt-incremental-model.md:63]: will find someone with some artistic skill to bring this character to life

File: docs/en/integrations/dbt/dbt-connecting.md (hunk 5 above)
Comment (gingerwizard) [reply docs/en/integrations/dbt/dbt-connecting.md:37]: I'd rather use screenshots, they are visually easier to consume imo. Will bump resolution

Comment (gingerwizard) [reply docs/en/integrations/dbt/dbt-connecting.md:37]: bumped the resolution

**Comment by gingerwizard** [comment]
@Ryado thanks for the suggestions, edits made.

**Review by melvynator — APPROVED** [review]

**Pull request merged**